drools-game-engine-horserace-kjar/src/main/resources/rules/game/gamerules.drl

```diff
@@ -0,0 +1,76 @@
+package org.drools.game.horserace;
+import org.drools.game.core.api.*;
+import org.drools.game.model.api.*;
+import org.drools.game.horserace.model.*;
+
+global GameMessageService messageService;
+global GameCallbackService callback;
+
+rule "Greeter"
+    when
+        $player : Player()
+    then
+        insert(  messageService.newGameMessage(  "Welcome, " + $player.getName() + ". Venture forth to the stables and mount a mighty steed!" ) );
+end
+
+rule "Give the player a team"
+    when
+        $player : Player()
+        not( Team( name == $player.name) ) 
+    then
+        Team team = new Team($player.getName());
+        team.addPlayer($player.getName());
+        insert(team);
+end
+
+rule "first checkpoint"
+        when
+            $player : Player()
+            $team : Team(name == $player.name && points == 0)
```

Comment:
If we remove the points evaluation here, we can add another generic rule to check for how many checkpoints the player has been before, and if those checkpoints were in order. Because now we have one model per game, I should rename the Room class here to Checkpoint, and the field playersInRoom can be removed/replaced by another mechanism to check which players already hit that checkpoint.